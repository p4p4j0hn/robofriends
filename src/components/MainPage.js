import React, { Component } from 'react';

import SearchBox from './SearchBox';
import Scroll from './Scroll';
import CardList from './CardList';
import ErrorBoundary from './ErrorBoundary';
import Header from './Header';
import './MainPage.css';

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = (robots) => {
    return robots.filter((robot) => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    });
  };

  render() {
    const { onSearchChange, robots, isPending } = this.props;

    return isPending ? (
      <div className='tc'>
        <Header />
        <h1 className='f2-ns'>Loading</h1>
      </div>
    ) : (
      <div className='tc wrapper'>
        <div className='header'>
          <Header />
          <SearchBox searchChange={onSearchChange} />
        </div>
        <Scroll>
          <ErrorBoundary fallback={<h2>Oooops. That is not good.</h2>}>
            <CardList robots={this.filteredRobots(robots)} />
          </ErrorBoundary>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
