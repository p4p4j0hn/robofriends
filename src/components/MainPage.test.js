import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

describe('test MainPage', () => {
  it('renders MainPage without crashing', () => {
    expect.assertions(1);
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: false
    };
    const wrapper = shallow(<MainPage {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('filters robots correctly', () => {
    expect.assertions(1);
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'John',
          email: 'john@gmail.com'
        }
      ],
      searchField: 'John',
      isPending: false
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filteredRobots()).toStrictEqual([
      {
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
      }
    ]);
  });

  it('filters robots correctly 2', () => {
    expect.assertions(1);
    const mockProps3 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'John',
          email: 'john@gmail.com'
        }
      ],
      searchField: 'a',
      isPending: false
    };
    const wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper3.instance().filteredRobots()).toStrictEqual([]);
  });

  it('tests isPending', () => {
    expect.assertions(1);
    const mockProps4 = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: true
    };
    const wrapper4 = shallow(<MainPage {...mockProps4} />);
    expect(wrapper4.find('[id="loading"]')).toHaveLength(1);
  });
});
