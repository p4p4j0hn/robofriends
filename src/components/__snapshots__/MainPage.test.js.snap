// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MainPage renders MainPage without crashing 1`] = `
<div
  className="tc wrapper"
>
  <div
    className="header"
  >
    <Header />
    <SearchBox />
  </div>
  <Scroll>
    <ErrorBoundary
      fallback={
        <h2>
          Oooops. That is not good.
        </h2>
      }
    >
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
